test/Oscoin/Test/P2P/IO.hs

```diff
@@ -49,6 +52,22 @@ propFramedHappy :: Property
 propFramedHappy = property $
     forAll nonEmptyFrobs >>= compatible framedServer framedClient
 
+propHybridHappy :: Property
+propHybridHappy = property $ do
+    framed    <- forAll nonEmptyFrobs
+    streaming <- forAll nonEmptyFrobs
+
+    annotate $
+           show (length framed) <> " framed, "
+        <> show (length streaming) <> " streaming"
+
+    xs' <-
+        map fst . evalIO . bind $ \(port, sock) ->
+            concurrently (hybridServer (length framed) sock) $
+                hybridClient port framed streaming
+
+    Just (framed <> streaming) === nonEmpty xs'
```

Comment:
Tip (well, depending on the interpretation 😁 ). In one of my recent PRs I did provided a drop-in replacement for `===` living in `Test.QuickCheck.Extended` which uses callstacks to pinpoint the failure but also uses my `Condensed` typeclass to allow for nicer outputs. Not sure it's your cup of 🍵 , but just wanted to flag up such this exists 😁

Reply:
This is actually using `hedgehog` :grimacing: But still a useful thing to know!

Reply:
oops 😬